### PR TITLE
Processor

### PR DESCRIPTION
**Pull request by Christian-B**
Adds a Processor

### CODE REVIEW

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                    "clockSpeed parameter " + clockSpeed
+                    + " cannot be less than or equal to zero");
+        }
+        if (dtcmAvailable <= 0) {
+            throw new SpinnMachineInvalidParameterException(
+                    "dtcmAvailable parameter " + dtcmAvailable
+                    + " cannot be less than or equal to zero");
+        }
+        return new Processor(processorId, MachineDefaults.CLOCK_SPEED,
+            MachineDefaults.DTCM_AVAILABLE, isMonitor);
```
Comment (dkfellows) [inline SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java]: These two lines above are wrong. They don't use the `clockSpeed` or `dtcmAvailable` argument.

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+     * Obtain a Processor Object for this ID which could be a monitor.
+     *
+     * @param processorId ID of the processor in the chip.
+     * @param isMonitor  Determines if the processor is considered the
+     *      monitor processor, and so should not be otherwise allocated.
+     * @return A default Processor Object with this ID and monitor state
+     */
+    public static final Processor factory(int processorId, boolean isMonitor) {
+        if (isMonitor) {
+            if (NON_MONITOR[processorId] == null) {
```
Comment (dkfellows) [inline SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java]: Finding monitor processors in a non-monitor array?

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java (hunk 1 above)
Comment (Christian-B) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java]: hides head in shame.

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java (hunk 2 above)
Comment (Christian-B) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Processor.java]: good thing head already hidden.

**Pull request merged**